### PR TITLE
add exception to business time rule, based on message characteristics

### PR DESCRIPTION
**Pull request by AnPogrebnyak**
subj

### CODE REVIEW

File: src/main/java/com/icthh/xm/tmf/ms/communication/rules/businesstime/BusinessTimeRule.java [hunk 1]
```diff
@@ -33,19 +36,57 @@ public RuleResponse validate(CommunicationMessage message) {
...
         return ruleResponse;
     }
+
+    /**
+     * Check whether the message contains characteristics which are allow to send it at not business time
+     * @param message Message to check
+     * @param businessDayConfig Configuration
+     * @return Whether the message contains characteristics which are allow to send it at not business time
+     */
+    private boolean checkMessageCharacteristics(CommunicationMessage message, BusinessDayConfig businessDayConfig) {
```
Comment (amedvedchuk) [inline src/main/java/com/icthh/xm/tmf/ms/communication/rules/businesstime/BusinessTimeRule.java]: suggest to rename `checkMessageCharacteristics` --> `isSendingAllowedByCharacteristic`

File: src/main/java/com/icthh/xm/tmf/ms/communication/rules/businesstime/BusinessTimeRule.java [hunk 2]
```diff
@@ -33,19 +36,57 @@ public RuleResponse validate(CommunicationMessage message) {
...
+            .getExceptionCharacteristics();
+
+        if (exceptionCharacteristics != null && !exceptionCharacteristics.isEmpty() && message.getCharacteristic() != null) {
+            exceptionCharacteristicPresent = message.getCharacteristic().stream()
+                .anyMatch(exceptionCharacteristics::contains);
+
+            if (exceptionCharacteristicPresent && log.isDebugEnabled()) {
+                // Map message to one string for Kibana and like a Json array for pretty look
+                log.debug(message.getCharacteristic().stream().filter(exceptionCharacteristics::contains)
+                    .map(
```
Comment (amedvedchuk) [inline src/main/java/com/icthh/xm/tmf/ms/communication/rules/businesstime/BusinessTimeRule.java]: Do you really need a JSON as a format of characteristics log output? Json concatenation logic seems too complex. 
Could you consider to collect `characteristic` to Map<String, Object> (key = characteristic.getName(), value = characteristic.getValue()) and just print this map?

Comment (AnPogrebnyak) [reply src/main/java/com/icthh/xm/tmf/ms/communication/rules/businesstime/BusinessTimeRule.java]: In the message these objects are the List, so what a purpose to display it in the log as map?  The developers will become confused  at the log analyzing.

File: src/main/java/com/icthh/xm/tmf/ms/communication/rules/businesstime/BusinessTimeRule.java (hunk 1 above)
Comment (AnPogrebnyak) [reply src/main/java/com/icthh/xm/tmf/ms/communication/rules/businesstime/BusinessTimeRule.java]: Done

File: src/main/java/com/icthh/xm/tmf/ms/communication/rules/businesstime/BusinessTimeRule.java (hunk 2 above)
Comment (amedvedchuk) [reply src/main/java/com/icthh/xm/tmf/ms/communication/rules/businesstime/BusinessTimeRule.java]: @AnPogrebnyak i was confused by constructing JSON with string concatenation. 
as a result output will be like 
`[{"name"=x, "value"=y}, {...}]`
With proposed Map solution output would be like 
`{x=y, ...}` which is almost the same but printing code is much simpler.

**Pull request merged**